source/common/router/rds_impl.cc

```diff
@@ -116,10 +117,23 @@ void RdsRouteConfigSubscription::onConfigUpdate(
     config_info_ = {new_hash, version_info};
     route_config_proto_ = route_config;
     stats_.config_reload_.inc();
-    ENVOY_LOG(debug, "rds: loading new configuration: config_name={} hash={}", route_config_name_,
-              new_hash);
-    for (auto* provider : route_config_providers_) {
-      provider->onConfigUpdate();
+
+    if (!route_config_proto_.has_vhds()) {
```

Comment:
Nit: prefer to have positive vs. negative logic as the conditional in `if {} else {}`

Reply:
fixed